tools/log-collector/log_collector.go

```diff
@@ -748,3 +714,132 @@ func (l *LogCollector) getResourceObjectsWithOwnerRef(resourcePath string,
 	}
 	return objects
 }
+
+func (l *LogCollector) getClusterNodes() error {
+
+	var nodeList unstructured.UnstructuredList
+	gvk := schema.GroupVersionKind{Kind: NodeKind, Version: CoreGv}
+	nodeList.SetGroupVersionKind(gvk)
+	err := l.k8sClient.List(context.Background(), &nodeList)
+	if err != nil {
+		log.Error("Unable to Fetch Node List")
+		return err
+	}
+	for _, node := range nodeList.Items {
+		resourceDir := filepath.Join(node.GetKind())
+		yErr := l.writeYaml(resourceDir, node)
+		if yErr != nil {
+			log.Error("Unable to Write Node Yaml")
+			return yErr
+		}
+	}
+
+	return nil
+}
+
+// WriteNs writes yaml of requested namespaces
+func (l *LogCollector) WriteNs(namespaceObjects unstructured.UnstructuredList) error {
+	for _, obj := range namespaceObjects.Items {
+		resourceDir := filepath.Join(obj.GetKind())
+		if !l.Clustered {
+			if contains(l.Namespaces, obj.GetName()) {
+				eLrr := l.writeYaml(resourceDir, obj)
+				if eLrr != nil {
+					return eLrr
+				}
+			}
+		} else {
+			eLrr := l.writeYaml(resourceDir, obj)
+			if eLrr != nil {
+				return eLrr
+			}
+		}
+	}
+	return nil
+}
+
+// checkNamespaces taken all given ns from user and checks the same in cluster and write it YAML's
+func (l *LogCollector) checkNamespaces() error {
```

Comment:
checkNamespaces should be a separate function to just check if the namespaces exist and should be called at the start. If the given namespace list is not correct then we should error out at the very beginning